app.js

```diff
@@ -458,7 +463,7 @@ solidityAPI.on("shapeAdded" , async (shapeAddress, ownerAddress) => {
      * We make a new shape
      */
 
-    const shapeData = await shapeContractData(shapeAddress);
+    const shapeData = shapeContractData(shapeAddress);
```

Comment:
Since shapeContractData is an `async` function, without the `await` here the `shapeData` variable will store a Promise. Is that intentional?